src/network/networkd-dhcp-server.c

```diff
@@ -298,6 +388,15 @@ int dhcp4_server_configure(Link *link) {
                 if (r < 0)
                         return r;
         }
+
+        LIST_FOREACH(vendor_specifics, p, link->network->static_dhcp_vendor_specifics) {
+                r = sd_dhcp_server_add_vendor_specific(link->dhcp_server, p->vendor_specific);
```

Comment:
Rather than putting `vendor_specific`, putting `type`, `length` and `data` should be fine. Then `sd_dhcp_server_set_vendor_specific_data()` or friends can be dropped, and `struct sd_dhcp_vendor_specific` can be internal, that is, can be moved to `.c` file.

Reply:
The idea is to expose `struct sd_dhcp_vendor_specific` data so that we can store this in `Network` object and stop duplicating memory similar to the `prefixes_by_section`